README.md
<h1>Simple and Stylish Image Slider</h1>
SnSiS was experimental project, but now i decided to develop it for newbies. It is easy to use. The good feature of SnSiS is you may customize it as you wish easily.

<h1>How to use</h1>
<hr>
Load SnSiS plugin after jQuery in your web page. then use <code>$("#wrapper").snsis();</code> to create events on elements. now you have installed SnSiS plugin. but in following i describe its options that you need for customizing.

<h2>[options]</h2>

<h6><font color="#ccc">fadingSpeed</font></h6>
Images fade in and fade out, it has set to <code>400</code> as default, but you can set integer or set <code>slow</code> or <code>fast</code> string.

<h6><font color="#ccc">imagesContainer</font></h6>
The element that consist of images for sliding. it has set to <code>#container</code> as default.

<h6><font color="#ccc">keyboardArrows</font></h6>
This plugin get you sliding using keyboard left and right arrows. but you can disable it by giving <code>false</code> to this option.

<h6><font color="#ccc">fullscreen</font></h6>
By clicking on images the fullscreen of the activated image will be run as default, but you can disable it by giving <code>false</code> to this option.


CHANGELOG.md
Check out what new is.

<h1>SnSiS Versions</h1>
<h2>0.1</h2>
<hr>
<ul>
<li>Croping images</li>
<li>Created plugin</li>
<li>Added new images in example file</li>
</ul>

<h2>1.1</h2>
<hr>
<ul>
<li>Changed images appeared in example file</li>
<li>Updated version of jQuery used in example file <font color="#ccc">(v1.9.0)</font></li>
<li>Optimized example file</li>
<li>Using Entypo icon font <font color="#ccc">(source: http://weloveiconfonts.com)</font></li>
<li>Rewriting plugin</li>
<li>Added resize icon to container</li>
<li>Disabled keyboard arrows (if enabled already) on resized image</li>
</ul>


js/snsis.js
(function ($) {
  $.fn.snsis = function (options) {
		var settings = $.extend({
			fadingOutSpeed: "fast",
			fadingInSpeed: "fast",
			imagesContainerElement: "article",
			enableKeyboardArrows: true,
			enableFullScreen: true,
		}, options),
			This = this,
			methods = {
				sliding: function () {

					$thisDataAction = $(this).data("action");
					active = $(settings['imagesContainerElement'] + " > img.active").index();

					$(settings['imagesContainerElement'] + " > img").eq(active).fadeOut(settings["fadingOutSpeed"], function () {
						$(this).removeClass("active");

						if (active == 0) active = 0;
						else if ($(settings['imagesContainerElement'] + " > img").size() - 1 == active) active = -1;

						if ($thisDataAction == "previous") active--;
						else if ($thisDataAction == "next") active++;

						$(settings['imagesContainerElement'] + " > img").eq(active).addClass("active").fadeIn(settings["fadingInSpeed"]);
					});


				},
				click: function () {
					methods["sliding"].apply(this);

				}
			};

		this.each(function () {

			$this = $(this);
			//checking elements and their attributes, and entered options
			if (typeof $this.data("action") === "undefined") {
				$.error("There is no 'data-action' attribute for controllers");
				return false;
			} else if (["previous", "next"].indexOf($this.data("action")) == -1) {
				$.error("'data-action' attribute must be 'previous' or 'next'");
				return false;
			} else if ($(settings['imagesContainerElement']).length == 0) {
				$.error("There is no " + settings['imagesContainerElement'] + " element");
				return false;
			}

			//binding click event on controllers
			$this.click(function () {
				methods["sliding"].apply(this);
			});
		});

		//binding left and right keyboard arrows on document
		if (settings['enableKeyboardArrows'] === true) {

			$(document).keydown(function (evt) {
				if (evt.keyCode == 37) methods["click"].apply($(This, "[data-action='previous']"));
				else if (evt.keyCode == 39) methods["click"].apply($(This, "[data-action='next']"));
			});
		}

		if (settings["enableFullScreen"]) {
			$("article > img").click(function (evt) {
				marginleft = '-' + ($(this).width() / 2) + 'px';
				marginTop = '-' + ($(this).height() / 2) + 'px';
				imgSrc = $(this).attr("src");
				$(".gray-window > div").html('<img src="' + imgSrc + '" style="margin-left:' + marginleft + ';margin-top:' + marginTop + '">');
				$(".gray-window").show();
			});

			$(".gray-window > div").click(function (evt) {
				evt.stopPropagation();
			});

			$(".gray-window").click(function () {
				$(".gray-window").hide();
			});

		}

	};
})(jQuery);


js/snsis-dev.js
(function ($) {
  $.fn.snsis = function (options) {
		var settings = $.extend({
			fadingOutSpeed: "fast",
			fadingInSpeed: "fast",
			imagesContainerElement: "article",
			enableKeyboardArrows: true,
			enableFullScreen: true,
		}, options),
			This = this,
			controllersElement = $(this).selector;
			methods = {
				sliding: function () {

					$thisDataAction = $(this).data("action");
					$imagesContainerElementNode = $(settings['imagesContainerElement'])[0];
					active = $("img.active",$imagesContainerElementNode).index();
					$("img",$imagesContainerElementNode).eq(active).fadeOut(settings["fadingOutSpeed"], function () {
						$(this).removeClass("active");

						if (active == 0) active = 0;
						else if ($("img",$imagesContainerElementNode).size() - 1 == active) active = -1;

						if ($thisDataAction == "previous") active--;
						else if ($thisDataAction == "next") active++;

						$("img",$imagesContainerElementNode).eq(active).addClass("active").fadeIn(settings["fadingInSpeed"]);
					});


				},
				click: function () {
					methods["sliding"].apply(this);

				}
			};

		this.each(function () {

			$this = $(this);
			//checking elements and their attributes, and entered options
			if (typeof $this.data("action") === "undefined") {
				$.error("There is no 'data-action' attribute for controllers");
				return false;
			} else if ($.inArray($this.data("action"),["previous", "next"]) == -1) {
				$.error("'data-action' attribute must be 'previous' or 'next'");
				return false;
			} else if ($(settings['imagesContainerElement']).length == 0) {
				$.error("There is no " + settings['imagesContainerElement'] + " element");
				return false;
			}

			//binding click event on controllers
			$this.click(function () {
				methods["sliding"].apply(this);
			});
		});

		//binding left and right keyboard arrows on document
		if (settings['enableKeyboardArrows'] === true) {

			$(document).keydown(function (evt) {
				if (evt.keyCode == 37) methods["click"].apply($(controllersElement+"[data-action='previous']"));
				else if (evt.keyCode == 39) methods["click"].apply($(controllersElement+"[data-action='next']"));
			});
		}

		if (settings["enableFullScreen"]) {
			$("article > img").click(function (evt) {
				marginleft = '-' + ($(this).width() / 2) + 'px';
				marginTop = '-' + ($(this).height() / 2) + 'px';
				imgSrc = $(this).attr("src");
				$(".gray-window > div").html('<img src="' + imgSrc + '" style="margin-left:' + marginleft + ';margin-top:' + marginTop + '">');
				$(".gray-window").show();
			});

			$(".gray-window > div").click(function (evt) {
				evt.stopPropagation();
			});

			$(".gray-window").click(function () {
				$(".gray-window").hide();
			});

		}

	};
})(jQuery);


js/snsis-min.js
(function($){$.fn.snsis=function(options){var settings=$.extend({fadingOutSpeed:"fast",fadingInSpeed:"fast",imagesContainerElement:"article",enableKeyboardArrows:true,enableFullScreen:true,},options),This=this,controllersElement=$(this).selector;methods={sliding:function(){$thisDataAction=$(this).data("action");$imagesContainerElementNode=$(settings['imagesContainerElement'])[0];active=$("img.active",$imagesContainerElementNode).index();$("img",$imagesContainerElementNode).eq(active).fadeOut(settings["fadingOutSpeed"],function(){$(this).removeClass("active");if(active==0)active=0;else if($("img",$imagesContainerElementNode).size()-1==active)active=-1;if($thisDataAction=="previous")active--;else if($thisDataAction=="next")active++;$("img",$imagesContainerElementNode).eq(active).addClass("active").fadeIn(settings["fadingInSpeed"])})},click:function(){methods["sliding"].apply(this)}};this.each(function(){$this=$(this);if(typeof $this.data("action")==="undefined"){$.error("There is no 'data-action' attribute for controllers");return false}else if($.inArray($this.data("action"),["previous","next"])==-1){$.error("'data-action' attribute must be 'previous' or 'next'");return false}else if($(settings['imagesContainerElement']).length==0){$.error("There is no "+settings['imagesContainerElement']+" element");return false}$this.click(function(){methods["sliding"].apply(this)})});if(settings['enableKeyboardArrows']===true){$(document).keydown(function(evt){if(evt.keyCode==37)methods["click"].apply($(controllersElement+"[data-action='previous']"));else if(evt.keyCode==39)methods["click"].apply($(controllersElement+"[data-action='next']"))})}if(settings["enableFullScreen"]){$("article > img").click(function(evt){marginleft='-'+($(this).width()/2)+'px';marginTop='-'+($(this).height()/2)+'px';imgSrc=$(this).attr("src");$(".gray-window > div").html('<img src="'+imgSrc+'" style="margin-left:'+marginleft+';margin-top:'+marginTop+'">');$(".gray-window").show()});$(".gray-window > div").click(function(evt){evt.stopPropagation()});$(".gray-window").click(function(){$(".gray-window").hide()})}}})(jQuery);


js/snsis.dev.js
/*
 * Simple and Stylish Image Slider
 * https://github.com/cyletech/SnSiS/
 * 
 * Copyright 2012 Alireza Eskandarpour Shoferi
 * Under the GNU GPLv3 license.
 * http://www.gnu.org/licenses/gpl-3.0.html
 */ (function ($) {
	$.fn.snsis = function (options) {
		var settings = $.extend({
			'fadingSpeed': 400,
			'imagesContainer': '#container',
			'controller': '#controllers',
			'keyboardArrows': true,
			'fullscreen': true,
		}, options),
			fromHere = $(settings['imagesContainer'] + ' > img.active').index(),
			imagesLength = $(settings['imagesContainer'] + ' > img').length;
		var methods = {
			slide: function (action) {
				var activeIndex = ($(settings['imagesContainer'] + ' > img.active').index()-1);
				if (action == 'next') {
					if (activeIndex == (imagesLength - 1)) fromHere = 0;
					else fromHere++;
				}
				else if (action == 'previous') {
					if (activeIndex == 0) fromHere = (imagesLength - 1);
					else fromHere--;
				}
				$(settings['imagesContainer'] + ' > img.active').fadeOut(settings['fadingSpeed']).removeAttr('class');
				$(settings['imagesContainer'] + ' > img').eq(fromHere).fadeIn(settings['fadingSpeed']).addClass('active');
			}
		};

		//bind click event on controllers
		$(settings['controller'] + ' > span', this[0]).click(function () {
			if ($(this).data('do') == 'next') methods['slide'].apply(this, ['next']);
			else if ($(this).data('do') == 'previous') methods['slide'].apply(this, ['previous']);
		});

		//keyboard arrows feature
		if (settings['keyboardArrows']) {
			$(document).keydown(function (e) {
				if ($('#gray-window').css('display') == 'none') {
					if (e.keyCode == 39) methods['slide'].apply(this, ['next']);
					else if (e.keyCode == 37) methods['slide'].apply(this, ['previous']);
				}
			});
		}

		//fullscreen feature
		if (settings['fullscreen']) {
			$(settings['imagesContainer'] + ' > img').click(function () {
				$('#gray-window > div').css({
					'width': $(this).width(),
					'height': $(this).height(),
					'margin-top': Math.abs($(this).height() / 2) * -1,
					'margin-left': Math.abs($(this).width() / 2) * -1
				}).html('<img src="' + $(this).attr("src") + '">').click(function (e) {
					e.stopPropagation();
				}).parent().click(function () {
					$(this).hide();
				}).show();
			});

			$(settings['imagesContainer']).prepend('<span title="Resize" class="icon entypo-resize-full"></span>');
		}
	}
})(jQuery);